network-mux/test/Test/Mux/ReqResp.hs

```diff
@@ -77,26 +77,25 @@ data TraceSendRecv msg
   deriving Show
 
 
-runDecoderWithChannel :: forall s m a.
+runDecoderWithChannel :: forall m a.
                          MonadST m
-                      => (forall b. ST s b -> m b)
-                      -> Channel m
+                      => Channel m
                       -> Maybe LBS.ByteString
-                      -> Decoder s a
+                      -> Decoder (PrimState m) a
```

Comment:
ugh, that's an ugly consequence of `stToIO`.  I am a bit surprised that's the only place where we needed to change it.

Reply:
Me too!